sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -39,29 +42,362 @@ class ConfigurationSetting(Model):
         "tags": {"key": "tags", "type": "{str}"},
     }
 
+    kind = "Generic"
+    content_type = None
+
     def __init__(self, **kwargs):
         super(ConfigurationSetting, self).__init__(**kwargs)
-        self.etag = kwargs.get("etag", None)
         self.key = kwargs.get("key", None)
         self.label = kwargs.get("label", None)
-        self.content_type = kwargs.get("content_type", None)
         self.value = kwargs.get("value", None)
+        self.etag = kwargs.get("etag", None)
+        self.content_type = kwargs.get("content_type", self.content_type)
         self.last_modified = kwargs.get("last_modified", None)
         self.read_only = kwargs.get("read_only", None)
-        self.tags = kwargs.get("tags", None)
+        self.tags = kwargs.get("tags", {})
```

Comment:
Why we give default value {}?

Given the case user creates a ConfigSetting w/o tags, read it and send it back, will this change the value of tag?

Reply:
No this will not.